core/trino-main/src/test/java/io/trino/sql/planner/assertions/FunctionCallProvider.java

```diff
@@ -138,7 +139,8 @@ public boolean equals(Object object)
 
             return Objects.equals(name, other.getName()) &&
                     other.getWindow().isPresent() &&
-                    Objects.equals(frame, other.getWindow().get().getFrame()) &&
+                    other.getWindow().get() instanceof WindowSpecification &&
```

Comment:
This Provider is used for plan assertions involving `WindowNode`. The provided `FunctionCall` is a way of describing what `WindowNode.Function` is expected. Because `FunctionCall` and `WindowNode.Function` are different abstractions, the Provider is constructed with `Frame` and not with `Window` (other window properties do not belong in `WindowNode.Function`). 

The `equals` method should never be called on `ExpectedWindowFunctionCall`. I don't see a purpose for it and it's not even symmetric.
Also, the Frame passed to the Provider is never accessed or compared with the actual value.

I suggest that we clean up this class as a follow-up. 

For now, I believe it is fair to adjust the `equals()` method the way I did, since the left-hand-side FunctionCall cannot have a `WindowReference`.